### PR TITLE
Add upstream xcframework improvements

### PR DESCRIPTION
**Pull request by keith**
See each commit for details. This depends on https://github.com/bazelbuild/bazel/commit/877845584186301e53ed01eaecf3166db4ddcec7 in bazel, which is not in the 5.x branches

### CODE REVIEW

**Review by thii — APPROVED** [review]

**Review by jpsim — APPROVED** [review]
`apple_static_xcframework` appears to be working well in my tests here: https://github.com/envoyproxy/envoy-mobile/pull/2217

File: apple/internal/apple_framework_import.bzl [hunk 1]
```diff
@@ -810,6 +810,8 @@ objc_library(
     ],
 )
 ```
+
+NOTE: This is only support on bazel 6.0+
```
Comment (jpsim) [inline apple/internal/apple_framework_import.bzl:814]:
```suggestion
NOTE: This is only supported on bazel 6.0+
```

Comment (keith) [reply apple/internal/apple_framework_import.bzl]: oof thanks

File: doc/rules-apple.md [hunk 2]
```diff
@@ -117,6 +117,8 @@ objc_library(
 )
 ```
 
+NOTE: This is only supported on bazel 6.0+
```
Comment (brentleyjones) [inline doc/rules-apple.md:120]: Does this only apply to `apple_static_framework_import`? What about `apple_static_xcframework`?

Comment (keith) [reply doc/rules-apple.md]: actually i think this one is fine, just not in our tests, because we generate one to tetst it

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**